tests/dead_test.py

```diff
@@ -301,6 +301,31 @@ def test_ignored_arguments(git_dir):
     assert not dead.main(())
 
 
+def test_regex_ignore_symbol(git_dir, capsys):
+    git_dir.join('f.py').write('_x = 1')
+    subprocess.check_call(('git', 'add', '.'))
+    assert dead.main(())
+    out, _ = capsys.readouterr()
+    assert out == '_x is never read, defined in f.py:1\n'
+    # now same with ignore symbol regex
+    assert not dead.main(('--exclude-symbol', '^_*'))
```

Comment:
`^_*` is probably not what you want, that matches `_` or `_asdfasfasdf` or `__________________________________asdfasdfasdf` -- you could simplify the regex to just `^_`

Reply:
actually, `^_*` also matches `asdfasdfadsf`

Reply:
First thanks for pointing out the typo in the regex, i meant `^_.*$` (symbols starting with underscore)

The idea is to mark symbols as intentionally "dead":
although `_, matching, _= "I could eat bananas all day".partition("all")`  is possible, i think
`_before, matching, _after = "I could eat bananas all day".partition("all")` is easier to read

You are right using an underscore masks all private functions as "undead" although they might be "dead", but having a regex gives the user to chose any pattern he/she wants.

Reply:
seems difficult to get correct and easy to get wrong so I'm leaning toward rejecting this